src/controllers/requestController.ts

```diff
@@ -135,6 +142,9 @@ export class RequestController {
 
             let previewUri = this.generatePreviewUri();
             ResponseStore.add(previewUri.toString(), response);
+            if (httpRequest.requestVariableCacheKey) {

```

Comment:
If the response is not successful according to the response status code, do we really need to save the response to cache? In other words, does the `RequestVariableCache` only saves the successful response.

Reply:
I don't know if this contributes, but I have encountered cases where extra information is returned in the content, for say a 404

Reply:
OK, @aphilpotts your advice is we sho uld keep existing logic that in some cases we need to reference the content in the error response

Reply:
Yes I've encountered cases where 4xx/5xx still returns content. Might actually be cases where people want to retrieve headers from error responses as well. I'd prefer to keep as-is.